.github/workflows/ubuntu.yml

```diff
@@ -0,0 +1,48 @@
+name: Ubuntu
+
+on:
+  push:
+    branches: [ dev ]
+
+env:
+  BUILD_TYPE: Release
+  INSTALL_LOCATION: .local
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    # if: "!contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[ci skip]')"
+
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: cache dependencies
+      uses: actions/cache@v2
+      id: cache
+      with:
+        path: ${{ github.workspace }}/${{ env.INSTALL_LOCATION }}
+        key: ${{ runner.os }}-dependencies
+
+    - name: install dependencies
+      if: ${{ steps.cache.output.cache-hit != 'true' }}
+      run: |
+          sudo apt-get update -y
+          sudo apt-get install -y libcmocka0
+          sudo apt-get install -y libcmocka-dev
+          sudo apt-get install -y cppcheck valgrind
+
+    - name: configure
+      run: cmake -Bbuild -DCMAKE_PREFIX_PATH=/usr/lib
+
+    - name: build
+      run: cmake --build build --config $BUILD_TYPE -j4
+
+    - name: run tests
+      run: |
+        cd build/test
+        valgrind ctest -VV
+
+    - name: Code coverage using Codecov 
+      run: bash <(curl -s https://codecov.io/bash) 
```

Comment:
{'detail': ErrorDetail(string='Could not find a repository, try using repo upload token', code='not_found')}